File: src/Kelvin.Server/Models/Preferences.cs
namespace Kelvin.Server.Models;

public enum TemperatureUnit
{
  Celsius,
  Fahrenheit,
}

public enum TimeFormat
{
  Hour24,
  Hour12,
}

public class Preferences : Entity
{
  public TemperatureUnit TemperatureUnit { get; set; }

  public TimeFormat TimeFormat { get; set; }

  public long? LocationId { get; set; }

  public string? LocationName { get; set; }
}

File: src/Kelvin.Client/app/models/preferences.ts
export type TemperatureUnit = 'Celsius' | 'Fahrenheit';

export type TimeFormat = 'Hour24' | 'Hour12';

export type Preferences = {
  temperatureUnit: TemperatureUnit;
  timeFormat: TimeFormat;
  locationId: number | null;
  locationName: string | null;
};
